### ISSUE TITLE
deprecation of current flutter version

### ISSUE THREAD

**Issue by Pranav108** [Saransh-cpp/BookRentApp-Chapter3#106]
# Description
I have upgraded my flutter version for my android studio few days ago now when I'm trying to run the code of this app then I'm getting a warning which suggest me to migrate the project to it's updated flutter version.

# Relevant files
`MainActivity.java/kt`
`android/app/src/main/AndroidManifest.xml`
`Drawable`
`styles.xml`

# Screenshot
<img src="https://user-images.githubusercontent.com/56934594/146627845-998f778f-41e0-4c31-a771-7c5342a82eee.png" alt="screenshot" width="800">

# References
[more_info](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects)

**Comment by Pranav108** [comment]
I never worked with migration of any app, but I would love to give it a try.
@Saransh-cpp Can I take it?

**Comment by Saransh-cpp** [comment]
Yes, go ahead!

**Comment by Saransh-cpp** [comment]
I might turn this into bonus if it requires too much work.

**Comment by Pranav108** [comment]
Do I have to remove this error only or have to migrate whole app?
Because after a single change in `AndroidManifest.xml` and updating the sdk  version removed this warning and code is running well.

**Comment by Saransh-cpp** [comment]
> Do I have to remove this error only or have to migrate whole app(like migration for null safety)?

This would be good for the long run. But if you want to get rid of the warning for now you can do so (would be an easy issue in that case).

**Comment by Pranav108** [comment]
> > Do I have to remove this error only or have to migrate whole app(like migration for null safety)?
> 
> This would be good for the long run. But if you want to get rid of the warning for now you can do so (would be an easy issue in that case).

Okay , I have just removed the warning and that was the only change I found from [this](https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects) resource.

**Comment by Saransh-cpp** [comment]
Re-opening this for complete migration

**Comment by rajoriaakash** [comment]
Hi,
Can I take this?

**Comment by rajoriaakash** [comment]
To migrate the app completely I must upgrade all dependencies into their null safety versions.
There is one dependency, carousel_pro that doesn't have a null-safety version.
Should I replace it with some other carousel dependency that ensures null safety?
And it would help a lot if the mentors could tell me which files am I supposed to look at, for changing the carousel code?
Thanks:)

**Comment by letsintegreat** [comment]
@rajoriaakash Yes, there are two dependencies that will cause problems with null safety. carousel_pro can be replaced with another dependency, while you will have to remove firebase_admob dependency completely, and also the code relevant to it.

I have also dealt with this issue in my PR, but it is yet to be reviewed. But fortunately, I created a separate commit while dealing with these two dependencies, so you can basically copy all the changes from this commit: https://github.com/Saransh-cpp/BookRentApp-Chapter3/pull/114/commits/7bf747b825986e3db9db4cd405980d924c2f042b

Additionally, there's this feature on Android Studio, where you can search for code across the entire project. The shortcut key is: Ctrl + Shift + F
So, you could just search for carousel_pro and it would show you all the relevant files. (Please ignore the files in node_modules.) But now you don't need to do that either, as you can basically copy my commit!

**Comment by rajoriaakash** [comment]
Oh okay!
Thanks for the detailed explanation :D
I will try my best :)

**Comment by rajoriaakash** [comment]
I did all the migration and stuff but now when I try to run the app.
The following error is encountered :
![image](https://user-images.githubusercontent.com/59030114/147873978-01b94f83-f537-4ece-81bf-2fc27327eb67.png)
I am not sure how this error appeared 😔
Here is my commit fe1d94333abaa626fa028ca078ebdc500631096a

**Comment by rajoriaakash** [comment]
All the migration changes have already been updated in the master branch by some other commit probably :).
This issue should be closed now
